sql/api/src/main/scala/org/apache/spark/sql/catalyst/parser/DataTypeAstBuilder.scala

```diff
@@ -76,11 +76,13 @@ class DataTypeAstBuilder extends SqlBaseParserBaseVisitor[AnyRef] {
       case (TIMESTAMP_LTZ, Nil) => TimestampType
       case (STRING, Nil) =>
         typeCtx.children.asScala.toSeq match {
-          case Seq(_) => SqlApiConf.get.defaultStringType
+          case Seq(_) =>
+            val collationId = SqlApiConf.get.defaultStringType.collationId
+            StringType(collationId, CollationStrength.Implicit)
```

Comment:
This probably relies on the change for session level default and DDL operations, but be careful of backwards compatibility.